src/ModelContextProtocol/Client/SseClientSessionTransport.cs

```diff
@@ -92,26 +92,25 @@ public override async Task SendMessageAsync(
         StreamableHttpClientSessionTransport.CopyAdditionalHeaders(httpRequestMessage.Headers, _options.AdditionalHeaders);
         var response = await _httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
 
-        response.EnsureSuccessStatusCode();
-
-        var responseContent = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
-
-        if (string.IsNullOrEmpty(responseContent) || responseContent.Equals("accepted", StringComparison.OrdinalIgnoreCase))
-        {
-            LogAcceptedPost(Name, messageId);
-        }
-        else
+        if (!response.IsSuccessStatusCode)
         {
             if (_logger.IsEnabled(LogLevel.Trace))
             {
-                LogRejectedPostSensitive(Name, messageId, responseContent);
+                LogRejectedPostSensitive(Name, messageId, await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false));
             }
             else
             {
                 LogRejectedPost(Name, messageId);
             }
 
-            throw new InvalidOperationException("Failed to send message");
+            response.EnsureSuccessStatusCode();
+        }
+
+        if (response.Content.Headers.ContentType?.MediaType is "application/json")
```

Comment:
Note that I've replaced the older heuristic (response content being empty or not "accepted") with a check against the media type.